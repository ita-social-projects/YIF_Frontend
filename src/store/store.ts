import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import dropboxReducer from '../store/reducers/dropboxReducer';
//import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import counterReducer from '../components/counter/counterSlice';
import requestStatusReducer from './reducers/requestStatus';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    dropbox: dropboxReducer,
    requestStatus: requestStatusReducer,
  },
});

export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
