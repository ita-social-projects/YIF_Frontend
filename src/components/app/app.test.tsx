import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from '../../store/store';
import App from './index';

test('fake test', () => {
  expect(true).toBeTruthy();
});

test('renders without crashing', () => {
  const app = render(
    <Provider store={store}>
      <App />
    </Provider>
  );

  expect(app).toMatchSnapshot();
});
