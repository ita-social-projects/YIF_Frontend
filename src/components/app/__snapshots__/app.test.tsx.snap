// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        Header
      </div>
      <div
        class="banner"
      >
        Banner
      </div>
      <div
        class="dropbox"
      >
        <div
          class="title_"
        >
          <h2>
            University filter
          </h2>
        </div>
        <form
          action="#"
          method="POST"
        >
          <div
            class="selectors"
          >
            <div
              class="box"
            >
              <ul
                class="dropboxMenu"
              >
                <li
                  class="input-li"
                >
                  <input
                    readonly=""
                    type="text"
                    value="Direction"
                  />
                  <ul
                    class="combolist-li"
                  >
                    <li>
                      Direction
                    </li>
                    <li>
                      d1
                    </li>
                    <li>
                      d2
                    </li>
                    <li>
                      d3
                    </li>
                    <li>
                      d4
                    </li>
                    <li>
                      d5
                    </li>
                    <li>
                      d6
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
            <div
              class="box"
            >
              <ul
                class="dropboxMenu"
              >
                <li
                  class="input-li"
                >
                  <input
                    readonly=""
                    type="text"
                    value="Speciality"
                  />
                  <ul
                    class="combolist-li"
                  >
                    <li>
                      Speciality
                    </li>
                    <li>
                      s1
                    </li>
                    <li>
                      s2
                    </li>
                    <li>
                      s3
                    </li>
                    <li>
                      s4
                    </li>
                    <li>
                      s5
                    </li>
                    <li>
                      s6
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
            <div
              class="box"
            >
              <ul
                class="dropboxMenu"
              >
                <li
                  class="input-li"
                >
                  <input
                    readonly=""
                    type="text"
                    value="University"
                  />
                  <ul
                    class="combolist-li"
                  >
                    <li>
                      University
                    </li>
                    <li>
                      u1
                    </li>
                    <li>
                      u2
                    </li>
                    <li>
                      u3
                    </li>
                    <li>
                      u4
                    </li>
                    <li>
                      u5
                    </li>
                    <li>
                      u6
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
            <button
              type="submit"
            >
              Search
            </button>
          </div>
        </form>
      </div>
      <div
        class="about"
      >
        About
      </div>
      <div
        class="cta"
      >
        Call to action
      </div>
      <div
        class="partners"
      >
        Partners
      </div>
      <div
        class="footer"
      >
        Footer
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      Header
    </div>
    <div
      class="banner"
    >
      Banner
    </div>
    <div
      class="dropbox"
    >
      <div
        class="title_"
      >
        <h2>
          University filter
        </h2>
      </div>
      <form
        action="#"
        method="POST"
      >
        <div
          class="selectors"
        >
          <div
            class="box"
          >
            <ul
              class="dropboxMenu"
            >
              <li
                class="input-li"
              >
                <input
                  readonly=""
                  type="text"
                  value="Direction"
                />
                <ul
                  class="combolist-li"
                >
                  <li>
                    Direction
                  </li>
                  <li>
                    d1
                  </li>
                  <li>
                    d2
                  </li>
                  <li>
                    d3
                  </li>
                  <li>
                    d4
                  </li>
                  <li>
                    d5
                  </li>
                  <li>
                    d6
                  </li>
                </ul>
              </li>
            </ul>
          </div>
          <div
            class="box"
          >
            <ul
              class="dropboxMenu"
            >
              <li
                class="input-li"
              >
                <input
                  readonly=""
                  type="text"
                  value="Speciality"
                />
                <ul
                  class="combolist-li"
                >
                  <li>
                    Speciality
                  </li>
                  <li>
                    s1
                  </li>
                  <li>
                    s2
                  </li>
                  <li>
                    s3
                  </li>
                  <li>
                    s4
                  </li>
                  <li>
                    s5
                  </li>
                  <li>
                    s6
                  </li>
                </ul>
              </li>
            </ul>
          </div>
          <div
            class="box"
          >
            <ul
              class="dropboxMenu"
            >
              <li
                class="input-li"
              >
                <input
                  readonly=""
                  type="text"
                  value="University"
                />
                <ul
                  class="combolist-li"
                >
                  <li>
                    University
                  </li>
                  <li>
                    u1
                  </li>
                  <li>
                    u2
                  </li>
                  <li>
                    u3
                  </li>
                  <li>
                    u4
                  </li>
                  <li>
                    u5
                  </li>
                  <li>
                    u6
                  </li>
                </ul>
              </li>
            </ul>
          </div>
          <button
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
    <div
      class="about"
    >
      About
    </div>
    <div
      class="cta"
    >
      Call to action
    </div>
    <div
      class="partners"
    >
      Partners
    </div>
    <div
      class="footer"
    >
      Footer
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
